src/compiler/scala/tools/nsc/backend/jvm/GeneratedClassHandler.scala

```diff
@@ -193,7 +195,10 @@ final case class CompilationUnitPaths(sourceFile: AbstractFile, outputDir: Abstr
  *   - Keeps a reference to the future that runs the post-processor
  *   - Buffers messages reported during post-processing
  */
-final class CompilationUnitInPostProcess(private var classes: List[GeneratedClass], val paths: CompilationUnitPaths) {
+final class CompilationUnitInPostProcess(
+  private var classes: List[GeneratedClass],
+  val sourceFile: AbstractFile,
+  val outputDir: AbstractFile) extends CompilationUnitPaths with BufferingBackendReporting {
```

Comment:
I think composition is cleaner here than inheritance. There's only one instance of this class per compilation unit, does it really matter for memory usage?